pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/SchemaInfoProvider.java

```diff
@@ -52,4 +54,13 @@ public interface SchemaInfoProvider {
      */
     String getTopicName();
 
+    /**
+     * Retrieve the configuration properties associated with this schema registry.
+     *
+     * @return schema registry configurations
+     */
+    default Map<String, String> getConfigs() {
```

Comment:
In addition to comment [here](https://github.com/gaoran10/pulsar/pull/33#discussion_r2244510189), passing the properties by `SchemaInfoProvider#getConfigs` is not clear.

Actually, we should not call `setSchemaInfoProvider` when the schema type is `External`.

Reply:
+1